lib/formatters/stringQuoteOnlyIfNecessary.js

```diff
@@ -0,0 +1,21 @@
+const os = require('os');
+const defaulStringFormatter = require('./string');
+
+function stringQuoteOnlyIfNecessaryFormatter(opts = {}) {
```

Comment:
Where is this formatter used?

Reply:
It's not used by default. But it's available if some user wants to use it since it solves #420 

I thought about having a single string formatter with many options. But having multiple formatters seems more maintainable to me.

Reply:
Excellent, just wanted to be sure.